### ISSUE TITLE
Problem in experiments

### ISSUE THREAD

**Issue by daidedou** [garvita-tiwari/PoseNDF#24]
Hi, thanks for releasing the pretrained model! I've downloaded it and launched the sample pose experiment. However, it doesn't work on my side (I didn't change anything). Moreover, the loss is stuck at 0.0056, and does not decrease after that. Do you know if I did something wrong? (I used version 2 of the model of course). This are the results I get (Init and output):
![init_0](https://user-images.githubusercontent.com/46010710/221907347-3df524c0-c505-49a2-9919-7db5086c5fa6.jpg)
![output_0](https://user-images.githubusercontent.com/46010710/221907381-458ff3bc-0668-458f-9b48-8c08f977fe12.jpg)
![init_1](https://user-images.githubusercontent.com/46010710/221907432-d1eb8909-d627-4fdc-aba0-8203ddbdd043.jpg)
![output_1](https://user-images.githubusercontent.com/46010710/221907443-2b576e44-11df-47f3-99db-1b73b80334a1.jpg)

**Comment by garvita-tiwari** [comment]
Are these poses generated by randomly sampling quaternions? For very weird poses, the network might not work, because such drastic poses were not seen in training. However, I still suspect that if you increase the number of steps, poses might get slightly better. If you wish to make the model work for this, then you need to train the network with such poses.

For this you can replace query pose(https://github.com/garvita-tiwari/PoseNDF/blob/main/data/prepare_traindata.py#L139) with this:

        quer_pose_quat = torch.rand((batch_size,21,4))
        quer_pose_quat = torch.nn.functional.normalize(quer_pose_quat,dim=2).to(device=device)
        quer_pose_np = quer_pose_quat.reshape(-1, 84).detach().cpu().numpy()

**Comment by daidedou** [comment]
Thank you, I think I get the idea. I tested indeed with more classic poses and it seemed to be much better. I will try training it by myself. Do you think I could use pretrained weights as a basis?

**Comment by garvita-tiwari** [comment]
> Thank you, I think I get the idea. I tested indeed with more classic poses and it seemed to be much better. I will try training it by myself. Do you think I could use pretrained weights as a basis?

Could you please explain, how do you want to use the pretrained weights as basis?

**Comment by daidedou** [comment]
Sorry, I wanted to initialize PoseNDF with your pre-trained weights and feed it with random poses in a new train loop. I was wondering if this would affect the behavior of the network in the case of close poses (sorry for the approximative English, and again, thank you for taking the time to answer me!).

**Comment by garvita-tiwari** [comment]
> Sorry, I wanted to initialize PoseNDF with your pre-trained weights and feed it with random poses in a new train loop. I was wondering if this would affect the behavior of the network in the case of close poses (sorry for the approximative English, and again, thank you for taking the time to answer me!).

That will probably not help. In our experiments, we find that training with random poses+ clean poses and then refining with noisy poses(created from AMASS) +clean poses helps. My guess is training in the opposite fashion will not help. But it's worth trying. You can still fine-tune with all the poses(instead of just random poses).I have trained a model with random poses and visualized some results here:
![init_0001](https://user-images.githubusercontent.com/25167952/223984185-f24e82f2-b709-40e5-b9da-9bbb080df2c2.jpg)
![out_0001](https://user-images.githubusercontent.com/25167952/223984197-a6d0252b-5652-42a6-a50e-62e542d9c395.jpg)
![init_0007](https://user-images.githubusercontent.com/25167952/223984265-5881f0f5-28f7-4d2f-a0d1-5f953b8748c1.jpg)
![out_0007](https://user-images.githubusercontent.com/25167952/223984280-4d447161-7ce8-4d08-9fae-7f8e9cac5dba.jpg)

You can find the corresponding model here: https://nextcloud.mpi-klsb.mpg.de/index.php/s/EdfTaLaiZindrCe
